### PR TITLE
DateTime Component: Fix sizing of help info

### PR DESCRIPTION
**Pull request by andrewserong**
## Description

This PR adds a couple of styling fixes to the DateTime component, to ensure that the help info view has an adequate minimum width, and that the focus styling on the Calendar Help button displays correctly.

* Add a minimum width to the calendar help view, since the popover's minimum width was removed in https://github.com/WordPress/gutenberg/pull/25218/
* Add a `z-index` to the buttons in the DateTime picker to ensure that when the Reset or Calender Help buttons are clicked/focused that the top border of the focus style is visible

## How has this been tested?

Manually, select the Publish date from the Status & visibility section in a post's settings panel.

1. Check that after clicking on the Calendar Help button, the border style of the button looks correct.
2. Check that the minimum width of the Calendar Help info area looks reasonable.

## Screenshots 

| Calendar Help info area before | Calendar Help info area after |
| --- | --- |
| image | image |

| Button focus style border visibility before | Button focus style border visibility after |
| --- | --- |
| image | image |

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested. (Manually tested)
- [x] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by andrewserong** [comment]
@jasmussen I've just added you as a reviewer since I see you've (fairly) recently done some work on this component 🙂

File: packages/components/src/date-time/style.scss [hunk 1]
```diff
@@ -46,6 +47,7 @@
 		height: 30px;
 		margin-top: 0;
 		margin-bottom: 0;
+		z-index: 1;
```
Comment (jasmussen) [inline packages/components/src/date-time/style.scss]: For `z-index: 1;` cases that simply elevate the item by one, I personally think it's fine to just do this. But I wanted to mention the z-index layer handling mixin we have. Was it intentional to not use it?

**Review by jasmussen — APPROVED** [review]
Yep, nice one.

Sorry I didn't catch this one when I initially removed the min-width, but ultimately I do think it's the responsibility of each of these components individually to set the min-width. So thanks for doing that!

**Comment by andrewserong** [comment]
Thanks for reviewing @jasmussen, I agree, it's better for the consumers of the component to define their own requirements for layout and sizing 👍

And thanks for mentioning the mixin, it wasn't intentional skipping using that — I saw there were a couple of other `z-index: 1` settings applied in this file, so I didn't think to look up the mixin.

I saw that there was an existing z-index layer for focused button components, so I've re-used that one instead of adding a separate one for this case. Let me know if you think it could do with tweaking further, otherwise I'll merge in this PR tomorrow 🙂

**Comment by jasmussen** [comment]
Thanks for the fix. I think it's always best practice to use the mixin, but especially in cases of "just elevate this by one" it can be okay to do without it. Mostly because z-index contexts aren't absolute and new contexts change them, the mixin makes most sense for those items that are all in the same stacking order.

It seems like you rebased at a bad time, tests are failing consistently on this one for some reason. Might want to try giving it a fresh one.

**Comment by andrewserong** [comment]
Thanks for the extra context Joen, this is all super helpful as I learn about more corners of the repo that I haven't explored before 😀

I've given this another rebase and all the CI actions are passing, so I'll merge it in (looks like the issue was resolved in https://github.com/WordPress/gutenberg/pull/34422). Thanks again!

**Pull request merged**